src/main/java/com/yelp/nrtsearch/server/luceneserver/field/PolygofieldDef.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import static com.yelp.nrtsearch.server.luceneserver.analysis.AnalyzerCreator.hasAnalyzer;
+
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.GeoPolygonQuery;
+import com.yelp.nrtsearch.server.luceneserver.doc.LoadedDocValues;
+import com.yelp.nrtsearch.server.luceneserver.field.properties.PolygonQueryable;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.LatLonShape;
+import org.apache.lucene.document.ShapeField;
+import org.apache.lucene.geo.Polygon;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.Query;
+
+public class PolygofieldDef extends IndexableFieldDef implements PolygonQueryable {
+
+  protected PolygofieldDef(String name, Field requestField) {
+    super(name, requestField);
+  }
+
+  @Override
+  protected void validateRequest(Field requestField) {
+    super.validateRequest(requestField);
+
+    if (requestField.getHighlight()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "field: %s cannot have highlight=true. only type=text or type=atom fields can have highlight=true",
+              getName()));
+    }
+
+    if (requestField.getStore()) {
+      throw new IllegalArgumentException("polygon fields cannot be stored");
+    }
```

Comment:
I don't understand this, aren't the polygons always going to be stored fields only ?

Reply:
This is same as ES https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java#L188.
In ES polygon does not support stored fields or doc value.(Only able to retrieve from _source)

Also we never retrieve this field in use cases and this field is pretty large. So I decide to make it not retrievable for now. I can change this later in a future pr

Reply:
I see that we don't actually add [StoredField](https://lucene.apache.org/core/8_4_0/core/org/apache/lucene/document/StoredField.html) for any type. Can leave it as is.